sale_order_create_event/models/sale_order.py

```diff
@@ -164,21 +164,28 @@ def _create_session_from_sale_line(
     def _prepare_session_data_from_sale_line(
             self, event, num_session, line, date):
         tz = self.env.user.tz
-        if line.performance:
-            if line.apply_performance_by_quantity:
-                duration = (line.performance * line.product_uom_qty)
+        new_date = False
+        duration = False
+        if self.project_id.working_hours:
+            new_date, duration = (
+                self.project_id.working_hours._calc_date_and_duration(date))
+        if not duration:
+            if line.performance:
+                if line.apply_performance_by_quantity:
+                    duration = (line.performance * line.product_uom_qty)
+                else:
+                    duration = line.performance
             else:
-                duration = line.performance
-        else:
-            duration = line.product_uom_qty
-        if line.project_by_task == 'yes':
-            utc_dt = _convert_to_utc_date(date, line.start_hour, tz=tz)
-        else:
-            utc_dt = _convert_to_utc_date(date, 0.0, tz=tz)
+                duration = line.product_uom_qty
+        if not new_date:
+            if line.project_by_task == 'yes':
```

Comment:
Y aqui lo mismo:
`utc_dt = _convert_to_utc_date(date, line.project_by_task == 'yes' and line.start_hour or 0.0, tz=tz)`